pkg/controller/volume/expand/expand_controller.go

```diff
@@ -221,11 +248,31 @@ func (expc *expandController) syncHandler(key string) error {
 		return err
 	}
 
-	return expc.expand(pvc, pv)
+	if volumePlugin.IsMigratedToCSI() {
```

Comment:
Even if the plugin support CSI migration, should also check for whether the migration is enabled and supported by kubelet using [`nodeUsingCSIPlugin` ](https://github.com/kubernetes/kubernetes/blob/master/pkg/volume/util/operationexecutor/operation_generator.go#L1745)

Reply:
Thats a good point but I do not think we can check if plugin is installed on the node via `nodeUsingCSIPlugin` because resize operation does not have a relationship with the node. 

Thinking some more I think even if CSI plugin was not installed on the node and control-plane expansion was done using CSI resizer, the file system expansion can still be performed by in-tree driver and everything should just work.

Reply:
Also, this will be less of a concern once https://github.com/kubernetes/enhancements/blob/master/keps/sig-storage/20190408-volume-scheduling-limits.md is impelemted, because then if CSI migration has been enabled for a plugin and plugin is not installed on the node, then the node will not have pod scheduled to the node.

I am thinking we may be able to remove entire `nodeUsingCSIPlugin` logic, once volume limit KEP is implemented.

Reply:
Node uses in-tree plugin while controller uses CSi driver is a combination we are trying to avoid in other cases. I remember is reason was it's hard to guarantee the compatibility between the two. 

@davidz627 How do you think?

Reply:
Sure - but:

a. We can't avoid it in this case.
b. Since nodes which doesn't have CSI plugin will not have migrated volumes scheduled on them, this will be a non-issue.

Reply:
My preference would be to keep the same logic we use for attach/mount sync here - that we want to use one type of backend for both controller calls & node calls and therefore we should use `nodeUsingCSIPlugin`. I understand that theoretically there shouldn't be an issue but that is also the case for *most* of the plugins for attach/mount steps as well. The issue is that it is impossible to actually guarantee that the controller/node level operations are compatible between migrated/not-migrated since there can be driver specific quirks.

@jsafrane I am working on a guide for requirements to CSI Driver authors.

Reply:
The problem is - contrary to attach/mount operation, no such relationship exists between node and control-plane resize operation. 

The volume may not be in-use by any pod when resize was attempted.

Reply:
There is an option for "delayed resizing" - wait until a pod is scheduled. But I don't like it, it breaks current behavior and looks ugly.

> The issue is that it is impossible to actually guarantee that the controller/node level operations are compatible between migrated/not-migrated since there can be driver specific quirks.

IMO, there are no driver specific quirks for filesystem resize in-tree. We can require the same behavior for migrated drivers.

Reply:
ok, understood!